packages/api/src/services/emp-state.ts

```diff
@@ -0,0 +1,162 @@
+import assert from "assert";
+import * as uma from "@uma/sdk";
+import Promise from "bluebird";
+const { emp } = uma.clients;
+import { BigNumber, utils } from "ethers";
+const { parseBytes32String } = utils;
+import { asyncValues, calcGcr } from "../libs/utils";
+import { Json, Libs } from "..";
+
+type Instance = uma.clients.emp.Instance;
+export default (config: Json, libs: Libs) => {
+  const { registeredEmps, provider, emps } = libs;
+
+  async function readEmpDynamicState(instance: Instance, address: string) {
```

Comment:
These make a _ton_ of requests at once. To get faster updates and decreased load on the node, I think we should probably use multicall to wrap all of this into a single on-chain query (and make this a standard pattern for situations like this).

Thoughts?

Reply:
great idea, would love to try this